packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -104,21 +109,24 @@ export abstract class AbstractAsyncHooksContextManager
 
     // patch methods that add a listener to propagate context
     ADD_LISTENER_METHODS.forEach(methodName => {
-      if (ee[methodName] === undefined) return;
-      ee[methodName] = this._patchAddListener(ee, ee[methodName], context);
+      const method = ee[methodName];
+      if (isGenericFunc(method)) {
```

Comment:
Same issue here, your type guard doesn't actually differentiate between this function and any other type of function.

Reply:
Answered above.